core/base/src/TError.cxx

```diff
@@ -131,17 +129,12 @@ void ErrorHandler(Int_t level, const char *location, const char *fmt, va_list ap
       fmt = "no error message provided";
 
    Int_t n = vsnprintf(buf, buf_size, fmt, ap);
-   // old vsnprintf's return -1 if string is truncated new ones return
-   // total number of characters that would have been written
-   if (n == -1 || n >= buf_size) {
-      if (n == -1)
-         buf_size *= 2;
-      else
-         buf_size = n+1;
+   if (n >= buf_size) {
+      buf_size = n+1;
       if (buf != &(small_buf[0])) delete [] buf;
       buf = 0;
       va_end(ap);
-      R__VA_COPY(ap, sap);
+      va_copy(ap, sap);
```

Comment:
```suggestion
      std::va_copy(ap, sap);
```
for consistency (of indicating we are using the c++ version).

Reply:
Unless I'm mistaken, `cstdio` does not put the functions/macros `va_copy` nor `va_end` in the `std` namespace.  Only the data type `va_list`.

Reply:
humm .. right :( ... they are still C macros  ....